### ISSUE TITLE
Where <depends> does not working properly while using after_element_html in fieldsets

### ISSUE THREAD

**Issue by SathishDreamorbit** [magento/magento2#18996]
### Preconditions (*)
1. Magento 2.1.9

### Steps to reproduce (*)
1.  Create a dependency fields in the admin grid 

```
$action = $fieldset->addField(
            'root',
            'select',
            [
                'name'      =>   'root',                
                'value' => '0',   
                'values'    =>      [
                    ['value' => 0,'label' => __("test1")],
                    ['value' => 1,'label' => __("test2")],
                ]
            ]
        );   
```                           

       $test1 = $fieldset->addField(
            'test1',
            'text',
            [
                'name' => 'test1',               
                 'disabled' => $isElementDisabled,
                'after_element_html' => '<label></label>'
            ]
        );

      $referenceLink =  $fieldset->addField(
            'test2',
            'text',
            [
                'name' => 'test2',
                'after_element_html' => '<label></label>'
            ]           
        );

### Expected result (*)

Need to work the dependency correctly.

### Actual result (*)

1. Hidden field label is showing
![image](https://user-images.githubusercontent.com/41613424/47853467-d3d53180-de04-11e8-8adb-e26d6f5a86fb.png)

**Comment by magento-engcom-team** [comment]
Hi @SathishDreamorbit. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [ ] Summary of the issue
- [ ] Information on your environment
- [ ] Steps to reproduce
- [ ] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me $VERSION instance` 

where `$VERSION` is [version tags](https://github.com/magento/magento2/tags) (starting from 2.2.0+) or develop branches (for example: 2.3-develop). 
For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@SathishDreamorbit do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [ ] yes
- [ ] no

**Comment by magento-engcom-team** [comment]
Hi @engcom-backlog-nazar. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [x] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [x] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [ ] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [ ] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [ ] 6. Add label `Issue: Confirmed` once verification is complete. 

- [ ] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by ghost** [comment]
Hi @SathishDreamorbit thank you for you report, this issue has already fixed in 2.2-develop branch 
only 2.2.0+ bug reports are considered nowadays. Try upgrading.

**Comment by SathishDreamorbit** [comment]
@engcom-backlog-nazar 

I understand that the issue is fixed in the Magento 2.2 version. Whereas my website is running on the version 2.1.9. We need the patches for the bugs which fixed.  So, that would be more helpful us to run the website smoothly.

**Comment by xuanmanhdao** [comment]
I am having the same problem in magento ver. 2.4.5-p1 . Please show me how to solve the problem
